### PR TITLE
[NL] Generate page config based on data availability and place relation

### PR DESCRIPTION
**Pull request by shifucun**
* Extend Tile to take optional `places` field so it can render from given places
* Put page config generation logic in func build_page_config(). Later on, this will extends to other query shape.

### CODE REVIEW

**Review by jehangiramjad — APPROVED** [review]
The Python changes look fine to me. Approving to unblock you but I'll let the others look at the tsx code :)

**Review by beets — APPROVED** [review]
very cool!

File: server/config/subject_page.proto [hunk 1]
```diff
@@ -113,6 +113,8 @@ message Tile {
   TileType type = 3;
   // Key for a StatVarSpec defined in the containing Category
   repeated string stat_var_key = 4;
+  // [optional] a list of place dcids used for this tile.
+  repeated string places = 7;
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/config/subject_page.proto]: Done

File: static/js/components/tiles/bar_tile.tsx [hunk 2]
```diff
@@ -41,6 +41,7 @@ interface BarTilePropType {
   id: string;
   title: string;
   place: NamedTypedPlace;
+  places: string[];
```
[the remark this answers is not in the document]
Comment (shifucun) [reply static/js/components/tiles/bar_tile.tsx]: Done, and updated to `comparison_places`

File: server/lib/nl_page_config.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  if spec.contained.svs:
+    block = category.blocks.add()
+    block.title = "{} in {}".format(spec.contained.contained_place_type,
+                                    spec.main.name)
+    column = block.columns.add()
+    tile = column
+    for sv in spec.contained.svs:
+      tile = column.tiles.add()
+      tile.type = subject_page_pb2.Tile.TileType.MAP
+      tile.title = sv2name[sv] + ': latest'
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/lib/nl_page_config.py:67]: Done

**Review by shifucun — COMMENTED** [review]
thanks for review!

**Pull request merged**